src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1241,26 +1241,29 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
             }
 
             // Add the content headers
-            if (request.Content != null)
+            if (request.Content == null)
+            {   
+                request.Content = new StringContent(string.Empty);
+                request.Content.Headers.Clear();
+            }
+
+            foreach (var entry in WebSession.ContentHeaders)
             {
-                foreach (var entry in WebSession.ContentHeaders)
+                if (SkipHeaderValidation)
                 {
-                    if (SkipHeaderValidation)
+                    request.Content.Headers.TryAddWithoutValidation(entry.Key, entry.Value);
+                }
+                else
+                {
+                    try
                     {
-                        request.Content.Headers.TryAddWithoutValidation(entry.Key, entry.Value);
+                        request.Content.Headers.Add(entry.Key, entry.Value);
                     }
-                    else
+                    catch (FormatException ex)
```

Comment:
It seems we haven't a test for the code branch.

Reply:
We do. This is not new code, just move outside of an if statement. 
example: https://github.com/PowerShell/PowerShell/blob/bf71b2c6161085ef12977fae88548b96cd93319d/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1#L944